javaparser-core/src/main/javacc/java.jj

```diff
@@ -673,6 +673,63 @@ TOKEN :
  * Program structuring syntax follows.
  */
 
+ /**
+ * The stub files are represented by the {@link StubUnit}
+ */
+StubUnit StubUnit():
+{
+	/* Represents the list of compilation units that contained in a stub file */
+	NodeList<CompilationUnit> compilationUnits = emptyList();
+	CompilationUnit cu;
+
+	/*
+	* Represents the particular compilation unit components in each iteration of
+	* reading compilation units from the stub file
+	*/
+	PackageDeclaration pakage = null;
+    NodeList<ImportDeclaration> imports = emptyList();
+    ImportDeclaration in = null;
+    NodeList<TypeDeclaration<?>> types = emptyList();
+    ModifierHolder modifier;
+    TypeDeclaration tn = null;
+    ModuleDeclaration module = null;
+}
+{
+	( LOOKAHEAD(2) ";" )*
+	(
+	    /*
+	    * Reading the particular compilation unit in the stub file
+	    * Number of compilation units could be 1+
+	    */
+        [ LOOKAHEAD(PackageDeclaration()) pakage = PackageDeclaration() ]
+        (
+            in = ImportDeclaration() { imports = add(imports, in); }
+            |
+            (
+                modifier = Modifiers()
+                (
+                    tn = ClassOrInterfaceDeclaration(modifier) { types = add(types, tn); }
+                |
+                    tn = EnumDeclaration(modifier) { types = add(types, tn); }
+                |
+                    tn = AnnotationTypeDeclaration(modifier) { types = add(types, tn); }
+                |
+                    module = ModuleDeclaration(modifier)
+                |
+                    ";"
+                )
+            )
+            {
+                 cu = new CompilationUnit(range(token_source.getHomeToken(), token()), pakage, imports, types, module);
+                 compilationUnits.add(cu);
+            }
+        )
```

Comment:
You have made a design choice to remove the `*` that appears at this location in CompilationUnit.  That affects how a file is split up:  you split it up into the largest possible number of CompilationUnits as opposed to the smallest possible number.  That's fine, but perhaps worth commenting about.